src/Dfc.CourseDirectory.WebV2/AddressSearch/GetAddressAddressSearchService.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Flurl;
+using Newtonsoft.Json;
+
+namespace Dfc.CourseDirectory.WebV2.AddressSearch
+{
+    public class GetAddressAddressSearchService : IAddressSearchService
+    {
+        private const string IdDelimiter = "::";
+
+        private readonly HttpClient _httpClient;
+        private readonly Options _options;
+
+        public GetAddressAddressSearchService(HttpClient httpClient, Options options)
+        {
+            _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+        }
+
+        public async Task<AddressDetail> GetById(string id)
+        {
+            if (string.IsNullOrWhiteSpace(id))
+            {
+                throw new ArgumentException($"{nameof(id)} cannot be null or whitespace.");
+            }
+
+            var idSegments = id.Split(IdDelimiter);
+
+            var results = await FindAddresses(idSegments?.FirstOrDefault() ?? throw new ArgumentException($"Failed to parse postcode from {nameof(id)}."));
```

Comment:
```suggestion
            var results = await FindAddresses(idSegments.FirstOrDefault() ?? throw new ArgumentException($"Failed to parse postcode from {nameof(id)}."));
```

Reply:
Will `idSegments?.FirstOrDefault()` ever be null?

Reply:
No, you're quite right.